velox/exec/RankLikeWindowBuild.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/exec/WindowBuild.h"
+
+namespace facebook::velox::exec {
+
+/// If the data has already been sorted according to the partition key and sort
+/// key, there is no need to hold all the data of a partition in memory for Rank
+/// and row_number functions. RankWindowBuild adopts a streaming method to
+/// construct WindowPartition, which can reduce the occurrence of Out Of Memory
+/// (OOM).
+class RankLikeWindowBuild : public WindowBuild {
```

Comment:
RankLike name doesn't work very well. dense_rank function is like rank, but it doesn't support this specific functionality. On the other hand, sum is not like rank, but it does support this functionality. Let's come up with a name that better reflects the functionality.

Reply:
@mbasmanova Yes. How about `RowLevelWindowBuild`?